packages/strapi-helper-plugin/lib/src/utils/Manager.js

```diff
@@ -0,0 +1,245 @@
+   * @returns {Object}
+   */
+  attrToRemoveInfos() {
+    return this.getAttrInfos(this.index);
+  }
+
+  /**
+   * 
+   * Retrieve the last element of each bootstrap line
+   * @returns {Array}
+   */
+  getLinesBound() { // NOTE: doesn't work for the last element if the line is not full!
+    const array = [];
+    let sum = 0;
+
+    this.list.forEach((item, i) => {
+      let { bootstrapCol, index, name, type } = this.getAttrInfos(i);
+
+      if (!type && name.includes('__col')) {
+        bootstrapCol = parseInt(name.split('__')[1].split('-')[2], 10);
+      }
+
+      sum += bootstrapCol;
+
+      if (sum === 12 || bootstrapCol === 12) {
+        const isFullSize = bootstrapCol === 12;
+        array.push({ name, index, isFullSize });
+        sum = 0;
+      }
+
+      if (sum > 12) {
+        sum = 0;
+      }
+
+      if (i < this.list.size - 1) {
+        let { bootstrapCol: nextBootstrapCol, name: nextName, type: nextType } = this.getAttrInfos(i + 1);
+        
+        if (!nextType && nextName.includes('__col')) {
+          nextBootstrapCol = parseInt(nextName.split('__')[1].split('-')[2], 10);
+        }
+
+        if (sum + nextBootstrapCol > 12) {
+          const isFullSize = bootstrapCol === 12;
+          array.push({ name, index, isFullSize });
+          sum = 0;
+        }
+      }
+    });
+
+    return array;
+  }
+
+  /**
+   * 
+   * Retrieve the field's type depending on its name
+   * @param {String} itemName 
+   * @returns {String}
+   */
+  getType(itemName) {
+    return this.state
+      .getIn(['modifiedSchema', 'models', ...this.keys, 'availableFields', itemName, 'type']);
+  }
+
+  /**
+   * Retrieve a field name depending on its index
+   * @param {Number} itemIndex
+   * @returns {String}
+   */
+  getAttrName(itemIndex){
+    return this.state
+      .getIn(['modifiedSchema', 'models', ...this.keys, 'fields', itemIndex]);
+  }
+
+  /**
+   * Retrieve the line bootstrap col sum
+   * @param {Number} leftBound 
+   * @param {Number} rightBound 
+   * @returns {Number}
+   */
+
+  getLineSize(elements) {
+    return elements.reduce((acc, current) => {
+      const appearance = this.layout.getIn([current, 'appearance']);
+      const type = appearance !== '' && appearance !== undefined ? appearance : this.getType(current);
+      const col = current.includes('__col') ? parseInt(current.split('__')[1].split('-')[2], 10) : this.getBootStrapCol(type);
+
+      return acc += col;
+    }, 0);
+  }
+
+  /**
+   * 
+   * @param {Bool} dir sup or min
+   * @param {Number} pivot the center
+   * @returns {Object} the first sup or last sup
+   */
+  getBound(dir, pivot = this.index) {
+    let result = {};
+    let hasResult = false;
+
+    this.arrayOfEndLineElements.forEach(item => {
+      const rightBondCond = findIndex(this.arrayOfEndLineElements, ['index', pivot]) !== -1 ? item.index < pivot : item.index <= pivot;
+      const cond = dir === true ? item.index >= pivot && !hasResult : rightBondCond;
```

Comment:
Complicated to read this one as well?

`item` is an element of the line?
`cond` is when an item isn't at the end of the line?
`dir` is the current direction?